module/criteria/helper.js

```diff
@@ -459,4 +468,80 @@ module.exports = class criteriaHelper {
     })
 }
 
+
+  /**
+   * Create copy of criterias
+   * @method
+   * @name createCopyOfCriterias
+   * @param {Array} criteriaIds - Array of criteria Id's        
+   * @returns {Object}  old and new Mapped criteria ids .  
+  */
+
+  static createCopyOfCriterias(criteriaIds= []) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        if (!criteriaIds.length) {
+          throw new Error(messageConstants.apiResponses.CRITERIA_ID_REQUIRED)
+        }
+
+        let criteriaDocuments = await this.criteriaDocument
+        (
+          { _id: { $in : criteriaIds }}
+        )
+
+        if (!criteriaDocuments.length) {
+           throw new Error(messageConstants.apiResponses.CRITERIA_NOT_FOUND)
+        }
+
+        let criteriaIdMap = {};
+        
+        await Promise.all(criteriaDocuments.map(async criteria => {
+          
+          await Promise.all(criteria.evidences.map(async evidence => {
+            await Promise.all(evidence.sections.map(async section => {
+              if (section.questions.length > 0) {
+                let criteriaQuestionIds = await questionsHelper.createCopyOfQuestions
+                  (
+                    section.questions
+                  )
+                if (criteriaQuestionIds.success && Object.keys(criteriaQuestionIds.data).length > 0) {
+                  section.questions = Object.values(criteriaQuestionIds.data);
+                }
+              }
+            }))
+          }))
+
+          criteria.externalId = criteria.externalId + "-" + gen.utils.epochTime();
+          let newCriteriaId = await database.models.criteria.create(_.omit(criteria, ["_id"]));
```

Comment:
@deepakr123 - Please store reference of source criteria in the new criteria

Reply:
resolved